### PR TITLE
llms/anthropic: adds full support for messages api

### LINKED ISSUE #695: anthropic: Support anthropic messages api
Anthropic has a more structured completion interface now, we should support it.

https://docs.anthropic.com/claude/reference/migrating-from-text-completions-to-messages

### PR DESCRIPTION
**Pull request by joeychilson**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [ ] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [ ] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

This adds full support for the new messages API including streaming. 
It backwards compatible with client option flag for using message api. 
It also updates the default model to 1.3 as 1.0 isn't supported by either completions or messages API.

Closes #695

### CODE REVIEW

**Comment by joeychilson** [comment]
I updated this to remove the fatals from the stream processing. I think it's better to pass the errors back instead of crashing and just letting the user handle them. This is different than the other LLM implementations like OpenAI which I used as a guideline.

File: llms/anthropic/internal/anthropicclient/messages.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+	if !ok {
+		return response, errors.New("invalid delta field type")
+	}
+	if stopReason, ok := delta["stop_reason"].(string); ok {
+		response.StopReason = stopReason
+	}
+
+	usage, ok := event["usage"].(map[string]interface{})
+	if !ok {
+		return response, errors.New("invalid usage field type")
```
Comment (tmc) [inline llms/anthropic/internal/anthropicclient/messages.go:275]: while an internal package, generally it's better to define these errors up top so calling code can use errors.Is checks.

**Review by tmc — APPROVED** [review]
Great contribution! some nits but they're for the internal client.

File: llms/anthropic/anthropicllm.go [hunk 2]
```diff
@@ -36,9 +43,10 @@ func New(opts ...Option) (*LLM, error) {
 
 func newClient(opts ...Option) (*anthropicclient.Client, error) {
 	options := &options{
-		token:      os.Getenv(tokenEnvVarName),
-		baseURL:    anthropicclient.DefaultBaseURL,
-		httpClient: http.DefaultClient,
+		token:             os.Getenv(tokenEnvVarName),
+		baseURL:           anthropicclient.DefaultBaseURL,
+		httpClient:        http.DefaultClient,
+		useCompletionsAPI: true,
```
Comment (tmc) [inline llms/anthropic/anthropicllm.go:49]: I'd actually like to default to the messages API

**Pull request merged**